android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/ContactsModule.java

```diff
@@ -187,408 +245,303 @@ public void getContactsAsync(final ReadableMap options, final Promise promise) {
     }
   }
 
-  private WritableMap addIdentityFromContentResolver(Set<String> fieldsSet, ContentResolver cr, WritableMap contact, long id) {
-    Cursor cursor = cr.query(
-        Data.CONTENT_URI,
-        null,
-        Data.CONTACT_ID + " = ? AND " + Data.MIMETYPE + " = ?",
-        new String[] { Long.toString(id), CommonDataKinds.StructuredName.CONTENT_ITEM_TYPE },
-        null
-    );
+  @NonNull
+  private Map<String, Contact> loadContactsFrom(Cursor cursor, Set<String> fieldsSet) {
 
-    if (cursor != null) {
-      Map<String, String> fields = Collections.unmodifiableMap(new HashMap<String, String>() {
-        {
-          put("firstName", CommonDataKinds.StructuredName.GIVEN_NAME);
-          put("lastName", CommonDataKinds.StructuredName.FAMILY_NAME);
-          put("name", CommonDataKinds.StructuredName.DISPLAY_NAME);
-          put("middleName", CommonDataKinds.StructuredName.MIDDLE_NAME);
-        }
-      });
-      Map<String, String> optional = Collections.unmodifiableMap(new HashMap<String, String>() {
-        {
-          put("phoneticFirstName", CommonDataKinds.StructuredName.PHONETIC_GIVEN_NAME);
-          put("phoneticLastName", CommonDataKinds.StructuredName.PHONETIC_FAMILY_NAME);
-          put("phoneticMiddleName", CommonDataKinds.StructuredName.PHONETIC_MIDDLE_NAME);
-          put("namePrefix", CommonDataKinds.StructuredName.PREFIX);
-          put("nameSuffix", CommonDataKinds.StructuredName.SUFFIX);
-        }
-      });
+    Map<String, Contact> map = new LinkedHashMap<>();
 
-      while (cursor.moveToNext()) {
-        for (String fieldName : fields.keySet()) {
-          String value = cursor.getString(cursor.getColumnIndex(fields.get(fieldName)));
-          if (value != null) {
-            contact.putString(fieldName, value);
-          }
-        }
-        for (String fieldName : optional.keySet()) {
-          if (fieldsSet.contains(fieldName)) {
-            String value = cursor.getString(cursor.getColumnIndex(optional.get(fieldName)));
-            if (value != null) {
-              contact.putString(fieldName, value);
-            }
-          }
-        }
+    while (cursor.moveToNext()) {
+      int columnIndex = cursor.getColumnIndex(ContactsContract.Data.CONTACT_ID);
+      String contactId;
+      if (columnIndex != -1) {
+        contactId = cursor.getString(columnIndex);
+      } else {
+        //todo - double check this, it may not be necessary any more
+        contactId = String.valueOf(-1);//no contact id for 'ME' user
```

Comment:
have you done some research about this? What actually is a 'me' user?

Reply:
> What actually is a 'me' user?
https://android.stackexchange.com/questions/17320/how-to-edit-the-me-contact

i honestly kept it as a safety carry over from react-native-contacts (https://github.com/rt2zz/react-native-contacts/pull/55/files)

i'll remove it, shouldn't cause trouble b/c up to this wouldn't have been relevant in expo api